packages/ts/react-i18n/src/backend.ts

```diff
@@ -1,17 +1,22 @@
-import type { Translations } from './types.js';
+import type { Translations, TranslationsResult } from './types.js';
 
 export interface I18nBackend {
-  loadTranslations(language: string): Promise<Translations>;
+  loadTranslations(language: string): Promise<TranslationsResult>;
 }
 
 export class DefaultBackend implements I18nBackend {
-  async loadTranslations(language: string): Promise<Translations> {
-    const url = `./?v-r=i18n&langtag=${language}`;
-    return fetch(url).then(async (response) => {
-      if (response.status === 200) {
-        return response.json();
+  async loadTranslations(language: string): Promise<TranslationsResult> {
+    return fetch(`./?v-r=i18n&langtag=${language}`).then(async (response) => {
+      if (!response.ok) {
+        return Promise.reject(new Error('Failed fetching translations.'));
       }
-      return Promise.reject(new Error(`Failed to load translations for language: ${language}`));
+      const retrievedLocale = response.headers.get('X-Vaadin-Retrieved-Locale');
+      const translations: Translations = await response.json().then((t) => t);
+      const translationsResult: TranslationsResult = {
+        translations,
+        resolvedLanguage: retrievedLocale ?? undefined,
+      };
+      return Promise.resolve(translationsResult);
     });
   }
```

Comment:
```suggestion
  async loadTranslations(language: string): Promise<TranslationsResult> {
    const response = await fetch(`./?v-r=i18n&langtag=${language}`);
    if (!response.ok) {
      throw new Error('Failed fetching translations.');
    }
    const retrievedLocale = response.headers.get('X-Vaadin-Retrieved-Locale');
    const translations: Translations = await response.json();
    const translationsResult: TranslationsResult = {
      translations,
      resolvedLanguage: retrievedLocale ?? undefined,
    };
    return translationsResult;
  }
```

Reply:
Pushed the commit. GitHub failed to commit the change from here directly.